src/relax/transform/fuse_tir.cc

```diff
@@ -391,10 +484,15 @@ class FusedTIRConstructor : public ExprVisitor {
       : mod_(mod), func_name_(func_name) {}
 
   void VisitExpr_(const FunctionNode* func) final {
+    auto relax_to_tir_var_map = RelaxToTIRVarMapCollector::Collect(mod_, GetRef<Function>(func));
     std::vector<Variant<tir::Var, tir::Buffer>> prim_func_params;
     for (const Var& relax_param : func->params) {
       size_t size_before = prim_func_params.size();
-      CollectPrimFuncParams(relax_param, &prim_func_params);
+      if (relax_to_tir_var_map.count(relax_param)) {
```

Comment:
Nitpick: Instead of having the conditional, both branches could be written as `CollectPrimFuncParams(relax_param, &prim_func_params, relax_to_tir_var_map.Get(relax_param))`.  The `Map::Get` method returns an `Optional<tir::Buffer>`, which is `NullOpt` when the key is absent from the map.

Reply:
Yes, that's a much better solution.